### PR TITLE
Fix internal links in Isolasi Mandiri page opens in a new tab

### LINKED ISSUE #535: Internal links in Isolasi Mandiri page opens in a new tab
## Description

On the stacked links in Isolasi Mandiri page, links that are meant to be internal opens in a new tab instead of in the current one.

image

## Expected Behaviour

Internal links (`e.g. FAQ`) should be opened in the same tab.

## Additional Information

The current implementation on `StackedLink` makes all links always open on a new tab, regardless of whether it is an external link or not. We should add a way to differentiate this, either through an extra Boolean item on the Netlify CMS page, or through a logic within the component.

### PR DESCRIPTION
**Pull request by ekamuktia**
Closes #535 

## Description

Check whether link is internal link by checking if the domain name contains "wargabantuwarga.com" (subdomain is also recognized as internal link) or whether it starts with `/` if the link is without domain name

## Current Tasks

- [x] Add internal link checking function 
- [x] Update StackedLink to open internal link in current tab
- [x] Add test cases

### CODE REVIEW

File: lib/__tests__/string-utils.test.ts [hunk 1]
```diff
@@ -92,3 +93,23 @@ describe("allIsEmptyString", () => {
...
+
+describe("isInternalLink", () => {
+  it.each`
+    input                                                                     | expected
+    ${"/faq"}                                                                 | ${true}
+    ${"https://www.wargabantuwarga.com"}                                      | ${true}
+    ${"https://www.wargabantuwarga.com/"}                                     | ${true}
+    ${"https://www.wargabantuwarga.com/provinces"}                            | ${true}
+    ${"https://www.wargabantuwarga.com/provinces"}                            | ${true}
+    ${"https://hotline.wargabantuwarga.com"}                                  | ${true}
```
Comment (mazipan) [inline lib/__tests__/string-utils.test.ts:105]: I think `https://hotline.wargabantuwarga.com` is not internal, it different subdomain.
It's safe to throw to new tab

Comment (ekamuktia) [reply lib/__tests__/string-utils.test.ts]: Updated. Let me know if it should be able to check link with no protocol (//www.wargabantuwarga.com), non http/https protocol, anchor tag  or other additional conditions.

File: lib/string-utils.ts [hunk 2]
```diff
@@ -113,3 +113,15 @@ export function getQueryParams(query: string): {} {
...
+
+export function isInternalLink(link: string): boolean {
+  const domain = link
+    .replace("https://", "")
+    .replace("http://", "")
+    .split("/")[0];
+  return (
+    link.startsWith("/") ||
+    domain == "www.wargabantuwarga.com" ||
+    domain == "wargabantuwarga.com"
```
Comment (resir014) [inline lib/string-utils.ts:125]: Always use `===`.

File: components/layout/stacked-link.tsx [hunk 3]
```diff
@@ -29,7 +30,7 @@ export default function StackedLink(list: StackedListProps) {
               <a
                 className="helper-link-cover"
                 data-testid={`next-link-${link.title}`}
-                target="_blank"
+                target={isInternalLink(link.url) ? "_self" : "_blank"}
```
Comment (resir014) [inline components/layout/stacked-link.tsx:33]: This does nothing when the `Link` component still wraps the entire `<a>` tag, which will override whatever's made here.

We need to modify this element to be wrapped with `Link` or not based on whether or not the URL is internal or external. See [here](https://github.com/kawalcovid19/wargabantuwarga.com/blob/main/components/layout/navigation-menu.tsx#L68-L84) for example.

**Review by resir014 — CHANGES_REQUESTED** [review]
This need more work, please take a look at the comments.

**Review by mazipan — APPROVED** [review]
Looks good to me

**Review by resir014 — APPROVED** [review]

File: lib/string-utils.ts (hunk 2 above)
Comment (zainfathoni) [reply lib/string-utils.ts]: Shall we enforce this rule using ESLint, @resir014?

**Pull request merged**